### ISSUE TITLE
CR creation fails

### ISSUE THREAD

**Issue by prary** [kiali/kiali#4001]
**Describe the bug**
Tried installing kiali via operator, but kiali creation fails with the following reason
Message:               Failed to create object: b'{"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure",
    "message":"roles.rbac.authorization.k8s.io \\"kiali\\" is forbidden: user \\"system:serviceaccount:t:kiali-operator\\"
     (groups=[\\"system:serviceaccounts\\" \\"system:serviceaccounts:t\\" \\"system:authenticated\\"]) 
     is attempting to grant RBAC permissions not currently held:\\n{APIGroups:[\\"\\"], Resources:[\\"nodes\\"], 
     Verbs:[\\"get\\" \\"list\\" \\"patch\\" \\"watch\\"]}\\n{APIGroups:[\\"\\"], 
     Resources:[\\"pods/log\\"], Verbs:[\\"patch\\"]}\\n{APIGroups:[\\"\\"], Resources:[\\"pods/proxy\\"], 
     Verbs:[\\"patch\\"]}","reason":"Forbidden","details":{"name":"kiali","group":"rbac.authorization.k8s.io","kind":"roles"},"code":403}\n'

**Versions used**
Kiali: v1.32.0
Istio: v1.8.4
Kubernetes flavour and version: (e.g. OpenShift Origin 3.9): v1.18.9

**To Reproduce**
Installed kiali operator via helm charts
Create kiali CR 

**Expected behavior**
Cr should be created without any error. Above error got fixed after adding CRUD on node resource https://github.com/kiali/helm-charts/blob/876bf94fb049be50d3ebfafd81174e9a63150cb5/kiali-operator/templates/clusterrole.yaml#L11 .

**Comment by jmazzitelli** [comment]
FYI we removed node access as it wasn't needed:

* https://github.com/kiali/helm-charts/commit/a3a7fb0ef0c5792587d512c2c4a7f96f9aa030bc
* https://github.com/kiali/kiali-operator/commit/1977882cb7b3f2fa923ece0eda3c57b89ee04885

I suspect you have a version mismatch between helm charts, operator, and or server somewhere. See my earlier comment and the PR that should help people avoid this in the future.

Workaround for you now, if you really have kiali operator v1.32, set the Kiali CR's `spec.deployment.image_version` to "v1.32"

**Comment by prary** [comment]
hi @jmazzitelli 
here is the kiali_cr.yaml
```
apiVersion: kiali.io/v1alpha1
kind: Kiali
metadata:
  name: kiali
  annotations:
    ansible.sdk.operatorframework.io/verbosity: "1"
spec:
#  installation_tag: ""

  istio_namespace: test

#  version: "default"
##########
#  ---
  auth:
    strategy: "anonymous"
#
##########
#  ---
  deployment:
    accessible_namespaces: ["^((?!(istio-operator|kube.*|openshift.*|ibm.*|kiali-operator)).)*$"]
#
#    ---
#    #additional_service_yaml:
#
    image_name: "xxx/kiali/kiali"
    image_version: v1.32.0
    ingress_enabled: true
    namespace: "test"
    override_ingress_yaml:
      metadata:
        annotations:
          nginx.ingress.kubernetes.io/secure-backends: "false"
          nginx.ingress.kubernetes.io/backend-protocol: "HTTP"
      spec:
        rules:
        - host: test.test
          http:
            paths:
            - path: /kiali
              backend:
                serviceName: kiali
                servicePort: 20001
# Defines compute resources that are to be given to the Kiali pod's container.
# The value is a dict as defined by Kubernetes. See the Kubernetes documentation
# https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container
#    ---
    resources:
      limits:
        cpu: 100m
        memory: 200Mi
      requests:
        cpu: 100m
        memory: 200Mi
    service_type: ClusterIP
#    verbose_mode: "3"
#    version_label: ""
    view_only_mode: true

##########
#  ---
#  external_services:
#
# Note about sensitive values in the external_services "auth" sections:
# Some external services configured below support an "auth" sub-section in order to tell Kiali how it should
# authenticate with the external services. Credentials used to authenticate Kiali to those external services can
# be defined in the "auth.password" and "auth.token" values within the "auth" sub-section.
# Because these are sensitive values, you may not want to declare the actual credentials here in the Kiali CR. In
# this case, you may store the actual password or token string in a Kubernetes secret. If you do, you need to
# set the "auth.password" or "auth.token" to a value in the format "secret:<secretName>:<secretKey>" where
# "<secretName>" is the name of the secret object that Kiali can access, and <secretKey> is the name of the key
# within the named secret that contains the actual password or token string. For example, if Grafana requires a
# password, you can store that password in a secret named "myGrafanaCredentials" in a key named "myGrafanaPw".
# In this case, you would set "external_services.grafana.auth.password" to "secret:myGrafanaCredentials:myGrafanaPw".
#
# **Custom-dashboards settings:
# discovery_auto_threshold: Threshold of the number of pods, for a given Application or Workload, above which dashboards discovery will be skipped
#   This setting only takes effect when discovery_enabled is set to 'auto'.
# discovery_enabled: Enable, disable or set 'auto' mode to the dashboards discovery process. If set to true, Kiali
#   will always try to discover dashboards based on metrics. Note that it can generate performance penalties while
#   discovering dashboards for workloads having many pods (thus many metrics).
#   When set to 'auto', Kiali will skip dashboards discovery for workloads with more than a configured threshold of pods
#   (see 'discovery_auto_threshold'). When discovery is disabled or auto/skipped, it is still possible to tie workloads
#   with dashboards through annotations on pods (refer to the doc https://kiali.io/documentation/latest/runtimes-monitoring/#pods-annotations)
#   Allowed values: true, false, auto.
# enabled: Enable or disable custom dashboards, including the dashboards discovery process. Default: true.
# is_core: Used in the Components health feature. When true, the unhealthy scenarios will be raised as errors. Otherwise, they will be raised as a warning.
# namespace_label: Prometheus label name used for identifying namespaces in metrics for custom dashboards.
#   Default is "kubernetes_namespace". It is quite common to use just "namespace" as well, depending on your Prometheus configuration.
# prometheus: Please check the section below about Prometheus-specific settings: they are identical. The Prometheus
#   configuration defined here is dedicated to fetching custom dashboards, hence allowing to use a different instance
#   of Prometheus. If omitted, the same Prometheus as for Istio metrics will be reused for custom dashboards.
#    ---
#    custom_dashboards:
#      discovery_auto_threshold: 10
#      discovery_enabled: auto
#      enabled: true
#      is_core: false
#      namespace_label: "kubernetes_namespace"
#      prometheus:
#        auth:
#          ca_file: ""
#          insecure_skip_verify: false
#          password: ""
#          token: ""
#          type: "none"
#          use_kiali_token: false
#          username: ""
#        health_check_url: ""
#        url: ""
#
# **Grafana-specific settings:
# auth: authentication settings to connect to Grafana:
#   ca_file: The certificate authority file to use when accessing Grafana using https. An empty string means no extra
#       certificate authority file is used. Default is an empty string.
#   insecure_skip_verify: Set true to skip verifying certificate validity when Kiali contacts Grafana over https.
#   password: Password to be used when making requests to Grafana, for basic authentication. User only requires viewer permissions. May refer to a secret - see note above.
#   token: Token / API key to access Grafana, for token-based authentication. It only requires viewer permissions. May refer to a secret - see note above.
#   type: The type of authentication to use when contacting the server from the Kiali backend. Use "bearer" to send the
#       token to the Grafana server. Use "basic" to connect with username and password credentials. Use "none" to not use any authentication.
#       Default is "none"
#   use_kiali_token: When true and if auth.type is "bearer", the same OAuth token used for authentication in Kiali will be used for the API calls to Grafana,
#       and auth.token config is ignored then.
#   username: Username to be used when making requests to Grafana, for basic authentication. User only requires viewer permissions.
# dashboards: A list of Grafana dashboards that Kiali can link to. Each item contains:
#   name: The name of the dashboard in Grafana
#   variables:
#     app: The name of a variable that holds the app name, if used in that dashboard (else it must be omitted)
#     namespace: The name of a variable that holds the namespace, if used in that dashboard (else it must be omitted)
#     service: The name of a variable that holds the service name, if used in that dashboard (else it must be omitted)
#     workload: The name of a variable that holds the workload name, if used in that dashboard (else it must be omitted)
# enabled: When true, Grafana support will be enabled in Kiali.
# health_check_url: Used in the Components health feature. This is the url which kiali will ping to determine whether the component is reachable or not. It defaults to `in_cluster_url` when not provided.
# in_cluster_url: Set URL for in-cluster access. Example: "http://grafana.istio-system:3000". This URL can contain query parameters if needed, such as "?orgId=1".
# is_core: Used in the Components health feature. When true, the unhealthy scenarios will be raised as errors. Otherwise, they will be raised as a warning.
# url: The URL that Kiali uses when integrating with Grafana. This URL must be accessible to clients external to
#      the cluster in order for the integration to work properly. If empty, an attempt to auto-discover it is made.
#      This URL can contain query parameters if needed, such as "?orgId=1".
#    ---
#    grafana:
#      auth:
#        ca_file: ""
#        insecure_skip_verify: false
#        password: ""
#        token: ""
#        type: "none"
#        use_kiali_token: false
#        username: ""
#      dashboards:
#      - name: "Istio Service Dashboard"
#        variables:
#          namespace: "var-namespace"
#          service: "var-service"
#      - name: "Istio Workload Dashboard"
#        variables:
#          namespace: "var-namespace"
#          workload: "var-workload"
#      - name: "Istio Mesh Dashboard"
#      - name: "Istio Control Plane Dashboard"
#      - name: "Istio Performance Dashboard"
#      - name: "Istio Wasm Extension Dashboard"
#      enabled: true
#      health_check_url: ""
#      in_cluster_url: "http://grafana.istio-system:3000"
#      is_core: false
#      url: ""
#
# **Istio-specific settings:
# component_status:
#   enabled: Enable/Disable of istio component status into masthead indicator. It defaults to true.
#   components: A list of components that Kiali will check its statuses.
#     app_label: Istio component pod app label.
#     is_core: Whether the component is core for your deployment.
#     namespace: The namespace where the component is installed in. It defaults to the 'istio_namespace' setting.
#                Note that the Istio documentation suggests you install the ingress and egress to different namespaces,
#                so you most likely will want to explicitly set this namespace value for the ingress and egress components.
#                For example, something like:
#                - app_label: istio-ingressgateway
#                  is_core:   false
#                  namespace: <the namespace where your ingress is deployed>
# config_map_name: The name of the istio control plane config map. It defaults to `istio`.
# envoy_admin_local_port: The port which kiali will open to fetch envoy config data information.
# istio_identity_domain: The annotation used by Istio to identify domains.
# istio_sidecar_injector_config_map_name: The name of the istio-sidecar-injector config map.
# istio_injection_annotation: The annotation used by Istio to automatically inject a specific workload
# istio_sidecar_annotation: The pod annotation used by Istio to identify the sidecar.
# istiod_deployment_name: The name of the istiod deployment.
# url_service_version: The Istio service used to determine the Istio version. If empty, assumes the URL for the well-known Istio version endpoint.
#    ---
#    istio:
#      component_status:
#        components:
#        - app_label: istiod
#          is_core: true
#        - app_label: istio-ingressgateway
#          is_core: true
#        - app_label: istio-egressgateway
#          is_core: false
#        enabled: true
#      config_map_name: "istio"
#      envoy_admin_local_port: 15000
#      istio_identity_domain: "svc.cluster.local"
#      istio_injection_annotation: "sidecar.istio.io/inject"
#      istio_sidecar_injector_config_map_name: "istio-sidecar-injector"
#      istio_sidecar_annotation: "sidecar.istio.io/status"
#      istiod_deployment_name: "istiod"
#      url_service_version: ""
#
#
# **Prometheus-specific settings:
# auth: authentication settings to connect to Prometheus:
#   ca_file: The certificate authority file to use when accessing Prometheus using https. An empty string means no extra
#       certificate authority file is used. Default is an empty string.
#   insecure_skip_verify: Set true to skip verifying certificate validity when Kiali contacts Prometheus over https.
#   password: Password to be used when making requests to Prometheus, for basic authentication. May refer to a secret - see note above.
#   token: Token / API key to access Prometheus, for token-based authentication. May refer to a secret - see note above.
#   type: The type of authentication to use when contacting the server from the Kiali backend. Use "bearer" to send the
#       token to the Prometheus server. Use "basic" to connect with username and password credentials. Use "none" to not use any authentication.
#       Default is "none"
#   use_kiali_token: When true and if auth.type is "bearer", Kiali Service Account token will be used for the API calls to Prometheus,
#       and auth.token config is ignored then.
#   username: Username to be used when making requests to Prometheus, for basic authentication.
# cache_duration: Prometheus caching duration expressed in seconds
# cache_enabled: Enable/disable Prometheus caching used for Health services
# cache_expiration: Prometheus caching expiration expressed in seconds
# health_check_url: Used in the Components health feature. This is the url which kiali will ping to determine whether the component is reachable or not. It defaults to `url` when not provided.
# is_core: Used in the Components health feature. When true, the unhealthy scenarios will be raised as errors. Otherwise, they will be raised as a warning.
# url: The URL used to query the Prometheus Server. This URL must be accessible from the Kiali pod.
#      If empty, assumes it is in the istio namespace at the URL "http://prometheus.<istio_namespace>:9090"
#    ---
#    prometheus:
#      auth:
#        ca_file: ""
#        insecure_skip_verify: false
#        password: ""
#        token: ""
#        type: "none"
#        use_kiali_token: false
#        username: ""
#      cache_duration: 10
#      cache_enabled: true
#      cache_expiration: 300
#      health_check_url: ""
#      is_core: true
#      url: ""
#
# **Tracing-specific settings:
#  - Right now we only support Jaeger
# auth: authentication settings to connect to Jaeger:
#   ca_file: The certificate authority file to use when accessing Jaeger using https. An empty string means no extra
#       certificate authority file is used. Default is an empty string.
#   insecure_skip_verify: Set true to skip verifying certificate validity when Kiali contacts Jaeger over https.
#   password: Password to be used when making requests to Jaeger, for basic authentication. User only requires viewer permissions. May refer to a secret - see note above.
#   token: Token / API key to access Jaeger, for token-based authentication. It only requires viewer permissions. May refer to a secret - see note above.
#   type: The type of authentication to use when contacting the server from the Kiali backend. Use "bearer" to send the
#       token to Jaeger Query. Use "basic" to connect with username and password credentials. Use "none" to not use any authentication.
#       Default is "none"
#   use_kiali_token: When true and if auth.type is "bearer", the same OAuth token used for authentication in Kiali will be used for the API calls to Jaeger Query,
#       and auth.token config is ignored then.
#   username: Username to be used when making requests to Jaeger, for basic authentication. User only requires viewer permissions.
# enabled: When true, connections to Jaeger are enabled. "in_cluster_url" and/or "url" need to be provided.
# health_check_url: Used in the Components health feature. This is the url which kiali will ping to determine whether the component is reachable or not. It defaults to `in_cluster_url` when not provided.
# in_cluster_url: Set URL for in-cluster access, which enables further integration between Kiali and Jaeger.
#      When not provided, Kiali will only show external links using the "url" config.
#      Example: "http://tracing.istio-system".
# is_core: Used in the Components health feature. When true, the unhealthy scenarios will be raised as errors. Otherwise, they will be raised as a warning.
# namespace_selector: Kiali use this boolean to look traces with namespace selector : service.namespace. Default: true
# url: External URL that will be used to generate links to Jaeger. It must be accessible to clients external to
#      the cluster (e.g: browser) in order to generate valid links.
#      If tracing service is deployed in a QUERY_BASE_PATH set this in the url like https://<hostname>/<QUERY_BASE_PATH> . EX: https://tracing-service:8080/jaeger
# use_grpc: Set "true" to enable GRPC connection between Kiali and Jaeger, in order to speed up the queries. In some setups you might not be able to use
#      GRPC (e.g. if Jaeger is behind some reverse proxy that doesn't support it).
#      If not specified, it will be false if deployed within a Maistra/OSSM+OpenShift environment, true otherwise.
# whitelist_istio_system: Set whitelist services in istio-system to check their traces
#    ---
#    tracing:
#      auth:
#        ca_file: ""
#        insecure_skip_verify: false
#        password: ""
#        token: ""
#        type: "none"
#        use_kiali_token: false
#        username: ""
#      enabled: true
#      health_check_url: ""
#      in_cluster_url: ""
#      is_core: false
#      namespace_selector: true
#      url: ""
#      #use_grpc:
#      whitelist_istio_system: ["jaeger-query", "istio-ingressgateway"]

##########
#  ---
#  health_config:
#
# rate: A list of health configurations that Kiali uses to determine what is (and is not) healthy nodes. Each item contains:
#   namespace: The name of the namespace that this configuration applies to. This is a regular expression.
#   kind: The type of resource that this configuration applies to. This is a regular expression.
#   name: The name of a resource that this configuration applies to. This is a regular expression.
#   tolerance: A list of tolerances for this configuration. Each item contains:
#     protocol: The protocol that applies for this tolerance (e.g. grpc or http). This is a regular expression.
#     direction: The direction that applies for this tolerance (e.g. inbound or outbound). This is a regular expression.
#     code: The status code that applies for this tolerance. This is a regular expression.
#     degraded: Health will be considered degraded when the telemetry reaches this value (specified as a %).
#     failure: A failure status will be shown when the telemetry reaches this value (specified as a %).
#    ---
#    rate: []

##########
#  ---
#  identity:
#
# Certificate file used to identify the file server. If set, you must go over https to access Kiali.
# The operator will set these if it deploys Kiali behind https.
# When left undefined, the operator will attempt to generate a cluster-specific cert file that provides
# https by default (today, this auto-generation of a cluster-specific cert is only supported on OpenShift).
# When set to an empty string, https will be disabled.
#    ---
#    #cert_file:
#
# Private key file used to identify the server. If set, you must go over https to access Kiali.
# When left undefined, the operator will attempt to generate a cluster-specific private key file that provide
# https by default (today, this auto-generation of a cluster-specific private key is only supported on OpenShift).
# When set to an empty string, https will be disabled.
#    ---
#    #private_key_file:

##########
#  ---
#  istio_labels:
#
# This section defines what labels Istio is using to indicate apps and versions.
# Typical values are: ("app" and "version") or ("app.kubernetes.io/name" and "app.kubernetes.io/version").
# Kiali needs to know what labels Istio is using to be in sync with what Istio considers applications.
# It adds the label used to instruct Istio to automatically inject sidecar proxies when applications are deployed.
#    ---
#    app_label_name: "app"
#    injection_label_name: "istio-injection"
#    version_label_name: "version"



##########
#  ---
#  server:
#
# Where the Kiali server is bound. The console and API server are accessible on this host.
#    ---
#    address: ""
#
# When true, allows additional audit logging on write operations.
#    ---
#    audit_log: true
#
# When true, allows the web console to send requests to other domains other than where the console came from.
# Typically used for development environments only.
#    ---
#    cors_allow_all: false
#
# When true, Kiali serves http requests with gzip enabled (if the browser supports it) when the requests are
#  over 1400 bytes.
#    ---
#    gzip_enabled: true
#
# When true, the metrics endpoint will be available for Prometheus to scrape.
#    ---
#    metrics_enabled: true
#
# The port that the server will bind to in order to receive metric requests.
# This is the port Prometheus will need to scrape when collecting metrics from Kiali.
#    ---
#    metrics_port: 9090
#
# The port that the server will bind to in order to receive console and API requests.
#    ---
#    port: 20001
#
# Defines the public domain where Kiali is being served. This is the "domain" part
# of the URL (usually it's a fully-qualified domain name).
# For example, "kiali.example.org".
# When empty, Kiali will try to guess this value from HTTP headers.
# On non-OpenShift clusters, you must populate this value if you want to enable
# cross-linking between Kiali instances in a multi-cluster setup.
#    ---
#    web_fqdn: ""
#
# Define the history mode of kiali UI. This can only take
# two possible values: either "browser" or "hash".
# When empty, it will always be considered as browser
#    ---
#    web_history_mode: ""
#
# Defines the ingress port where the connections come from. This is usually
# necessary when the application responds through a proxy/ingress, and it does
# not forward the correct headers so Kiali can guess the port.
#
# When empty, Kiali will try to guess this value from HTTP headers.
#    ---
#    web_port: ""
#
# Defines the context root path for the Kiali console and API endpoints and readiness probes.
# When providing a context root path that is not "/", do not add a trailing slash.
# For example, use "/kiali" not "/kiali/".
# When empty, will default to "/" on OpenShift and "/kiali" on Kubernetes.
#    ---
#    web_root: ""
#
# Defines the public HTTP schema used to serve Kiali. This can only take
# two possible values: either "http" or "https".
# When empty, Kiali will try to guess this value from HTTP headers.
# On non-OpenShift clusters, you must populate this value if you want to enable
# cross-linking between Kiali instances in a multi-cluster setup.
#    ---
#    web_schema: ""
```

kiali-operator yaml
```# Source: kiali-operator/templates/deployment.yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: RELEASE-NAME-kiali-operator
  namespace: test
  labels:
    helm.sh/chart: kiali-operator-0.0.0
    app: kiali-operator
    app.kubernetes.io/name: kiali-operator
    app.kubernetes.io/instance: RELEASE-NAME
    version: "0.0.0"
    app.kubernetes.io/version: "0.0.0"
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/part-of: "kiali-operator"
spec:
  replicas: 1
  selector:
    matchLabels:
      app.kubernetes.io/name: kiali-operator
      app.kubernetes.io/instance: RELEASE-NAME
  template:
    metadata:
      name: RELEASE-NAME-kiali-operator
      namespace: test
      labels:
        # required for the operator SDK metric service selector
        name: RELEASE-NAME-kiali-operator
        helm.sh/chart: kiali-operator-0.0.0
        app: kiali-operator
        app.kubernetes.io/name: kiali-operator
        app.kubernetes.io/instance: RELEASE-NAME
        version: "0.0.0"
        app.kubernetes.io/version: "0.0.0"
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/part-of: "kiali-operator"
      annotations:
        prometheus.io/scrape: "true"
    spec:
      serviceAccountName: RELEASE-NAME-kiali-operator
      containers:
      - name: operator
        image: "testkiali/kiali-operator:v1.32.0"
        imagePullPolicy: Always
        args:
        - "--zap-log-level=info"
        - "--leader-election-id=RELEASE-NAME-kiali-operator"
        volumeMounts:
        - mountPath: /tmp/ansible-operator/runner
          name: runner
        env:
        - name: WATCH_NAMESPACE
          value: ""
        - name: POD_NAME
          valueFrom:
            fieldRef:
              fieldPath: metadata.name
        - name: POD_NAMESPACE
          valueFrom:
            fieldRef:
              fieldPath: metadata.namespace
        - name: ALLOW_AD_HOC_KIALI_NAMESPACE
          value: "true"
        - name: ALLOW_AD_HOC_KIALI_IMAGE
          value: "false"
        - name: PROFILE_TASKS_TASK_OUTPUT_LIMIT
          value: "100"
        - name: ANSIBLE_DEBUG_LOGS
          value: "true"
        - name: ANSIBLE_VERBOSITY_KIALI_KIALI_IO
          value: "1"
        - name: ANSIBLE_CONFIG
          value: "/etc/ansible/ansible.cfg"
        ports:
        - name: http-metrics
          containerPort: 8080
      volumes:
      - name: runner
        emptyDir: {}
      affinity:
        {}
```

**Comment by jmazzitelli** [comment]
Looks like you have some customizations to the Kiali install, so I can't help. I don't know what customizations were made and I suspect something in the way you build and installed this is causing the problem

Notice:
```
        app.kubernetes.io/version: "0.0.0"
```
That tells me you are not using the published helm charts to install. Also:

```
        image: "testkiali/kiali-operator:v1.32.0"
```

I cannot vouch for this image. But I can tell you if it was based on v1.32.0 release, you have to make sure you are using the helm charts for the published v1.32 in order that the operator gets the appropriate permissions created for it. If you are using helm charts from master branch, things will break (most likely in the way you are seeing).

In short, my recommendation is to use the published v1.32 helm charts to install the operator and use the published kiali-operator v1.32 that is published on quay.io. If the published v1.32 release causes this problem, then we can investigate a potential bug in the permissions granted to the operator.

**Comment by lucasponce** [comment]
This one seems answered and no more info/data provided.
